src/adapters/scene.js

```diff
@@ -102,9 +116,30 @@ Scene.prototype.initMapBox = function() {
       this.popup.addListener(interactiveLayer);
     });
 
+    // we have to delay click event resolution to make time for possible double click events,
+    // which are thrown *after* two separate click events are thrown
+    this.clickDelayHandler = null;
     this.mb.on('click', e => {
       const pois = this.mb.queryRenderedFeatures(e.point, { layers: interactiveLayers });
-      this.clickOnMap(e.lngLat, pois[0]);
+      // when clicking on a POI, just trigger the action without delay,
+      // as a subsequent double click isn't a problem
+      if (pois[0]) {
+        this.clickOnMap(e.lngLat, pois[0]);
+        return;
+      }
+      // cancel the previous click handler if it's still pending
+      clearTimeout(this.clickDelayHandler);
```

Comment:
This should be moved called when clicking on a POI too, to avoid an edge case where the action of a first click (without interactive layer) would be triggered **after** the second click (on a POI)

Reply:
Nice find. Thank you!